File: ai-server/app/models/product_cluster_item.py
from sqlalchemy import Column, BigInteger, ForeignKey
from sqlalchemy import Float

from app.models.base import Base


class ProductClusterItem(Base):

    __tablename__ = "product_cluster_item"

    id = Column(BigInteger, primary_key=True)

    cluster_id = Column(
        BigInteger,
        ForeignKey("product_cluster.id")
    )

    product_id = Column(
        BigInteger,
        ForeignKey("product.id")
    )

    confidence_score = Column(Float)

File: api-server/src/main/java/com/productresearch/apiserver/domain/cluster/entity/ProductClusterItem.java
package com.productresearch.apiserver.domain.cluster.entity;

import com.productresearch.apiserver.domain.product.entity.Product;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "product_cluster_item")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
@AllArgsConstructor
public class ProductClusterItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cluster_id")
    private ProductCluster cluster;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private Product product;

    @Column(name = "confidence_score")
    private Double confidenceScore;
}